pkg/exchange/bitget/types.go

```diff
@@ -136,3 +138,121 @@ func (e *BookEvent) ToGlobalOrderBooks() []types.SliceOrderBook {
 
 	return books
 }
+
+type SideType string
+
+const (
+	SideBuy  SideType = "buy"
+	SideSell SideType = "sell"
+)
+
+func (s SideType) ToGlobal() (types.SideType, error) {
+	switch s {
+	case SideBuy:
+		return types.SideTypeBuy, nil
+	case SideSell:
+		return types.SideTypeSell, nil
+	default:
+		return "", fmt.Errorf("unexpceted side type: %s", s)
+	}
+}
+
+type MarketTrade struct {
+	Ts    types.MillisecondTimestamp
+	Price fixedpoint.Value
+	Size  fixedpoint.Value
+	Side  SideType
+}
+
+type MarketTradeSlice []MarketTrade
+
+func (m *MarketTradeSlice) UnmarshalJSON(b []byte) error {
```

Comment:
How about adding a guard to avoid `m==nil`?

Reply:
Great! I have added it . Avoid stepping on landmines in the future.